### DISCUSSION TITLE
I installed Command Line Tools from the Apple installer after running 'brew install gcc'. Did I mess up my setup?

### DISCUSSION THREAD

**Discussion by tomelam** [Homebrew/discussions#1402] [Getting started] [answered]
I am getting a subtle error when I run the installation check for Gambit Scheme (github.com/gambit/gambit) on macOS Big Sur (version 11.2.3) on an M1 MacBook. The same installation and check worked perfectly on macOS Catalina (version 10.15.7) on a late 2012 iMac.

The only thing I did that might have messed up my toolchain was that I stupidly installed Command Line Tools using Apple's installer after running 'brew install gcc'.

Did I mess up my system? If I did, can I uninstall gcc, then uninstall Command Line Tools, then run 'brew doctor', then reinstall gcc to fix things? Or something else?

To be clear: I am not getting any error that I can confidently blame on my silly use of Apple's Command Line Tools installer.

**Comment by SMillerDev** [comment]
> The only thing I did that might have messed up my toolchain was that I stupidly installed Command Line Tools using Apple's installer after running 'brew install gcc'.

Brew needs the CLT to function properly. This won't ever break something.

It's much more likely that something is breaking because you have one machine based on ARM and one on x86_64.

**Reply by tomelam** [reply]
So my 'double installation' really didn't mess up anything?

**Comment by carlocab** [comment, marked as answer]
How did you install `brew` on your M1 mac? Did you use the installer, or did you just copy over your Catalina install (perhaps using Migration Assistant)?

Note that the `gcc` bottle requires a CLT install, so uninstalling the CLT then reinstalling `gcc` will likely require you to reinstall the CLT using Apple's installer.

**Reply by tomelam** [reply]
I used the installer. I did not copy over my Catalina installation.

So my 'double installation' really didn't mess up anything?

**Reply by carlocab** [reply]
I'm not really sure what you mean by 'double installation' here. What did you install twice, and how? Do you mean you installed the CLT twice? (i.e. once when installing `brew` and then again by running Apple's installer after installing `gcc`)

**Reply by tomelam** [reply]
As I understand/remember it, 'brew install gcc' will install CLT. But after doing that, I downloaded and ran Apple's installer package, which would 'install CLT again'. That's what I meant by a 'double installation'. I think that explains what I did and how I did it.

And, yes, I 'installed CLT twice', once when installing brew's gcc formula and again by running Apple's installer after installing gcc.

**Reply by carlocab** [reply]
Ah, no, `brew install gcc` does not install the CLT. It will, however, fail to install (unless building from source) if you don't have the CLT installed.

The `brew` install script does install the CLT if you don't already have it installed, so you might have installed it twice. I don't think it hurts, however. If you're really concerned you can do
```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

**Reply by tomelam** [reply]
I'm only concerned because I've seen a lot of problems when I didn't follow blackbox scripts exactly right.

Thanks for your answer. It's worth it to me to follow the 'rm -rf' and 'xcode-select' procedure you mentioed. I'm marking your answer correct.

**Reply by tomelam** [reply]
Um, does that thumbs up count as marking a reply as the correct answer?

**Reply by SMillerDev** [reply]
Nope, the checkbox in the top right corner does, FTFY

**Reply by tomelam** [reply]
Thanks!

**Reply by tomelam** [reply]
**Postscript**
After I removed CLT, I just checked to see what 'brew install gcc' would do then. It just exited, saying 'No developer tools installed...' and mentioning the xcode-select command needed to be run.

I believe I forgot that I had seen the same message the first time I ran 'brew install gcc'. The answer marked 'correct' helped me to recall and understand what happened.

**Reply by carlocab** [reply]
Yup, that is exactly what I said would happen in https://github.com/Homebrew/discussions/discussions/1402#discussioncomment-688601.
